### PR TITLE
Bug sending messages issue

### PR DESCRIPTION
**Pull request by jaytranuts247**
- fixed messages display
- fixed chat user display
- fixed messages send function.
- fixed security issues

Let me know if there is any error or anything I need to do to improve

### CODE REVIEW

File: client/src/Login.js
Comment (dericf) [inline client/src/Login.js:10]: In future PRs, it is best if you only include files and changes that are directly related to the issue - this allows us to more better review your changes in isolation. This file is just formatting changes and updating deprecated props and does not need to be included. 🙂

File: client/src/components/Sidebar/Chat.js
Comment (dericf) [inline client/src/components/Sidebar/Chat.js:51]: Good catch, but let's try and keep the changes specific to the current issue. If you want to make additional fixes or improvements, you can save those for your optional PR 🙂

File: client/src/components/Sidebar/Sidebar.js
Comment (dericf) [inline client/src/components/Sidebar/Sidebar.js:52]: Same comment as above about only including relevant changes

File: client/src/socket.js
Comment (dericf) [inline client/src/socket.js:17]: Is this doing anything ? if not, it doesn't need to stay in the code, and we shouldn't commit console logs to our code either 🙂

File: client/src/store/utils/thunkCreators.js
Comment (dericf) [inline client/src/store/utils/thunkCreators.js:98]: Nice 💯

File: server/routes/api/conversations.js
Comment (dericf) [inline server/routes/api/conversations.js:26]: The problem with doing this (alone) is that now the order of the conversations is wrong, and the text that shows the most recent message now displays the last! Please update this solution to ensure the order of conversations and message preview text are not broken 🙂

File: client/src/socket.js
Comment (dericf) [inline client/src/socket.js:28]: Unfortunately this isn't the security fix that needed to be addressed, the issue is when a user posts a message to the server, we are making assumptions about the conversationId that gets passed in. 

See if you can add some security in the endpoint where messages get created - we want to make sure that users can only post messages to conversations the have access to 🔒

**Review by dericf — COMMENTED** [review]
Nice work on this first PR 👍 

Please address the inline comments and let us know in slack when you're ready for a second review

File: client/src/Login.js
Comment (jaytranuts247) [reply client/src/Login.js:10]: is it ok for me to leave it there for future PRs, or do you want me to revert it back to its original file ?

File: client/src/components/Sidebar/Chat.js
Comment (jaytranuts247) [reply client/src/components/Sidebar/Chat.js:51]: is it ok for me to leave it there for future PRs, or do you want me to revert it back to its original file ?

File: client/src/components/Sidebar/Sidebar.js
Comment (jaytranuts247) [reply client/src/components/Sidebar/Sidebar.js:52]: is it ok for me to leave it there for future PRs, or do you want me to revert it back to its original file ?

File: client/src/socket.js
Comment (jaytranuts247) [reply client/src/socket.js:17]: it will be removed.

File: client/src/store/utils/reducerFunctions.js
Comment (cassandrale179) [inline client/src/store/utils/reducerFunctions.js:17]: Is this the fix for the broke conversation order? I think it worked, but I'm curious if the runtime of this will be bad since we're doing a clone and it probably have to  iterate over all the messages to clone. An alternative way you can approach this is just to keep your original solution and reverse the conversation before we returned to the user.

**Review by cassandrale179 — APPROVED** [review]
Looks good overall but I left some comments to address. After that you can merged 💯

File: client/src/store/utils/reducerFunctions.js
Comment (jaytranuts247) [reply client/src/store/utils/reducerFunctions.js:17]: My solution worked fine. I changed flex to "column-reverse" to reverse the display other of the messages. Moreover, in #2 where you want to find the id of the latest messages that have been read. so you do not have to do a reverse map. just use map and the first item is the latest message you map through. 

Please let me know what you think.

File: server/db/models/conversation.js
Comment (jaytranuts247) [reply server/db/models/conversation.js:25]: noted !

File: client/src/store/utils/reducerFunctions.js
Comment (jaytranuts247) [reply client/src/store/utils/reducerFunctions.js:17]: The messages is sent from the server is in "DESC" order. so the latest message is in the beginning of the messages array. If you use push then new message message will be added in the end, which messing the order of the messages array. so the new message should be added at the start. so it would be fine there. And the only change is from "column-reverse" in Messages.js file

**Pull request merged**
